core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -136,10 +139,10 @@ public abstract void handleResolvedAddresses(
    * <p>The initial state of a Subchannel is IDLE. You won't get a notification for the initial IDLE
    * state.
    *
-   * <p>If a Subchannel has transitioned from a state that would be picked for RPCs, to a state that
-   * would not be picked, this method should create a new picker and call {@link
-   * Helper#updatePicker}, in order to prevent unnecessary delays of RPCs. Please refer to {@link
-   * PickResult#withSubchannel}'s javadoc for more information.
+   * <p>If the new state is not SHUTDOWN, this method should create a new picker and call {@link
```

Comment:
`CONNECTING` as well? I don't think most LBs would need to change anything for `IDLE`→`CONNECTING`... although I guess `TRANSIENT_FAILURE`→`CONNECTING` would be one, but it does go through `IDLE` first... Seems nuanced...